src/Backend/Modules/Pages/Engine/Model.php

```diff
@@ -101,195 +92,16 @@ public static function buildCache(string $language = null): void
-
-            // Insert new meta
-            $metaRepository->add($meta);
-            $metaRepository->save($meta);
-
-            // build page
-            $page = new Page(
-                $sourceData['id'],
-                BackendAuthentication::getUser()->getUserId(),
-                $sourceData['parent_id'],
-                $sourceData['template_id'],
-                $meta,
-                $toLanguage,
-                $sourceData['type'],
-                $sourceData['title'],
-                new DateTime(),
-                $sourceData['sequence'],
-                $sourceData['navigation_title_overwrite'],
-                $sourceData['hidden'],
-                Status::active(),
-                $sourceData['type'],
-                $sourceData['data'],
-                $sourceData['allow_move'],
-                $sourceData['allow_children'],
-                $sourceData['allow_edit'],
-                $sourceData['allow_delete']
-            );
-
-            // insert page, store the id, we need it when building the blocks
-            $pageRepository->add($page);
-            $pageRepository->save($page);
-
-            $revisionId = $page->getRevisionId();
-
-            $blocks = [];
-
-            // get the blocks
-            $sourceBlocks = self::getBlocks($activePage->getId(), null, $fromLanguage);
-
-            // loop blocks
-            foreach ($sourceBlocks as $sourceBlock) {
-                // build block
-                $block = $sourceBlock;
-                $block['revision_id'] = $revisionId;
-                $block['created_on'] = BackendModel::getUTCDate();
-                $block['edited_on'] = BackendModel::getUTCDate();
-
-                // Overwrite the extra_id of the old content block with the id of the new one
-                if (in_array($block['extra_id'], $contentBlockOldIds)) {
-                    $block['extra_id'] = $contentBlockIds[$block['extra_id']];
-                }
-
-                if (BackendModel::isModuleInstalled('Location')) {
-                    // Overwrite the extra_id of the old location widget with the id of the new one
-                    if (in_array($block['extra_id'], $locationWidgetOldIds)) {
-                        $block['extra_id'] = $locationWidgetIds[$block['extra_id']];
-                    }
-                }
-
-                // add block
-                $blocks[] = $block;
-            }
-
-            // insert the blocks
-            self::insertBlocks($blocks);
-
-            $text = '';
-
-            // build search-text
-            foreach ($blocks as $block) {
-                $text .= ' ' . $block['html'];
-            }
-
-            // add
-            BackendSearchModel::saveIndex(
-                'Pages',
-                (int) $page->getId(),
-                ['title' => $page->getTitle(), 'text' => $text],
-                $toLanguage
+        /** @noinspection NullPointerExceptionInspection */
```

Comment:
is there a way we can make this more generic? Like that it works everywhere at once
Because otherwise we'll end up filling fork with comments like this

Reply:
Blame Symfony :) Not sure to make this more generic.